### DISCUSSION TITLE
GitHub Actions Event Payload out of Date

### DISCUSSION THREAD

**Discussion by jtmullen** [community/community#25427] [Actions]
<p>Hello, I am running a custom github action on a private repo that has to reference the sha of the last commit on both the base branch and the head branch.</p>
<p>I have found that if the base branch is updated while the PR is open the event payload available to my action ends up out of date for the base sha. Pushes to the head branch update the head sha but the base sha stays the same as when it was opened. To me this doesn’t seem like intended behavior, but it seems weird that I am seeing one sha update and not the other.</p>
<p>In the action I can see in the actions/Checkout logs that it gets the most recent base branch:</p>
<pre><code class="lang-auto"> HEAD is now at &lt;merge sha&gt; Merge &lt;head_sha&gt; into &lt;base_sha&gt;
</code></pre>
<p><code>.pull_request.head.sha</code> is the same as the one above<br>
<code>.pull_request.base.sha</code> is an old one, from when the PR was opened.</p>
<p>I have opened a GitHub support ticket, since it is all in private repos, but I haven’t heard back in a week now so I wanted to see if anyone here might know why this is the case.  Thanks!</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/jtmullen">@jtmullen</a>,</p>
<p>Here is a ticket for the similar topic:</p><aside class="quote quote-modified" data-post="2" data-topic="138877">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/old-workflow-used-after-commit-change/138877/2">Old workflow used after commit change?</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows, now with world-class CI/CD.">GitHub Actions</span></a>
  </div>
  <blockquote>
    <a class="mention" href="/pcolmer">@pcolmer</a>, 

Since I’m using pull_request_target, my understanding is that Actions will use the workflow from the master branch. 

Yes, your understanding is correct. 
However, there are more things you also need to know: 


When opening a PR, the base commit SHA (github.event.pull_request.base.sha) generally will point to the latest commit on the target branch of the PR, if the source branch is up-to-date with the target branch. The base commit SHA normally is fixed and not automatically changed…
  </blockquote>
</aside>

<p>As mentioned in that ticket, you need to <strong>update the pull request branch</strong> to make it up-to-date with the base branch (target branch) of the PR.</p>

**Comment by jtmullen** [comment]
<p>Thanks, I heard back from support just now also. Seems I was a bit confused on what base_sha was representing.</p>
